src/pal/src/file/file.cpp

```diff
@@ -3605,18 +3721,26 @@ GetTempFileNameW(
             goto done;
         }
     }
-       
+    
+    tempfile_name = (CHAR *)alloca((MAX_LONGPATH+1) * sizeof(CHAR));
```

Comment:
Moreover, alloca never returns NULL, if there is not enough space on the stack, the behavior is undefined. So the alloca is the same as using the array here.